apps/src/code-studio/header.js

```diff
@@ -173,21 +190,72 @@ function shareProject() {
 
     ReactDOM.render(
       <Provider store={getStore()}>
-        <ShareDialog
+        <ConnectedShareDialog
           i18n={i18n}
           icon={appOptions.skin.staticAvatar}
           shareUrl={shareUrl}
           isAbusive={dashboard.project.exceedsAbuseThreshold()}
+          isOwner={dashboard.project.isOwner()}
+          isPublished={dashboard.project.isPublished()}
           channelId={dashboard.project.getCurrentId()}
           appType={appType}
           onClickPopup={popupWindow}
           // TODO: Can I not proliferate the use of global references to Applab somehow?
           onClickExport={window.Applab ? window.Applab.exportApp : null}
+          onShowPublishDialog={showPublishDialog}
+          onUnpublish={unpublishProject}
           canShareSocial={canShareSocial}
         />
       </Provider>,
       dialogDom
     );
+
+    getStore().dispatch(showShareDialog());
+  });
+}
+
+const ConnectedPublishDialog = connect(state => ({
+  isOpen: state.header.publishDialog.isOpen,
+  isPublishPending: state.header.publishDialog.isPublishPending,
+  projectId: state.header.publishDialog.projectId,
+  projectType: state.header.publishDialog.projectType,
+}), dispatch => ({
+  onClose: () => dispatch(hidePublishDialog()),
+}))(PublishDialog);
+
+function showPublishDialog() {
```

Comment:
If I understand what's happening here correctly (and it's quite possible that I don't), we have a ShareDialog component that based on some user action might want to later display a PublishDialog.

Rather than giving share dialog a callback that goes and creates a new DOM node and renders into it, I would expect that ShareDialog's render method has something like:
```
render() {
  return (
    <div>
       ...
       {stateThatSaysIWantPublishDialog && <PublishDialog/>}
    </div>
  );
}